awscli/argprocess.py

```diff
@@ -245,13 +245,36 @@ def _is_complex_shape(model):
     return True
 
 
+def _uses_old_list_case(service_name, operation_name, argument_name):
```

Comment:
This is used in both the shorthand parser and the shorthand doc generator, which are both in this module. I could put it as part of a super class for both of them to get it out of the module scope, though I'm not sure that makes much sense.